pilot/pkg/proxy/envoy/config.go

```diff
@@ -270,11 +280,21 @@ func buildSidecarListenersClusters(
 		// only HTTP outbound clusters are needed
 		httpOutbound := buildOutboundHTTPRoutes(mesh, node, instances, services, config)
 		httpOutbound = buildEgressHTTPRoutes(mesh, node, instances, config, httpOutbound)
-		clusters = append(clusters,
-			httpOutbound.clusters()...)
-		listeners = append(listeners,
-			buildHTTPListener(mesh, node, instances, nil, listenAddress, int(mesh.ProxyHttpPort),
-				RDSAll, useRemoteAddress, traceOperation, true, config))
+		clusters = append(clusters, httpOutbound.clusters()...)
+		opts := httpListenerOpt{
+			mesh:             mesh,
+			node:             node,
+			instances:        instances,
+			routeConfig:      nil,
+			ip:               listenAddress,
+			port:             int(mesh.ProxyHttpPort),
+			rds:              RDSAll,
+			useRemoteAddress: useRemoteAddress,
+			direction:        traceOperation,
+			outboundListener: true,
+			store:            config,
+		}
```

Comment:
neat.. This is exactly what I was telling @ijsnellf . A single context struct passed to buildXYZ, such that we get rid of the lengthy arg list.

Reply:
this was split into a separate PR as mentioned.